AWS Secret Engine: Support Session Tokens (#12735)
hashicorp/vault: website/content/docs/secrets/aws.mdx

    ---           -----
    access_key    AKIA3ALIVABCDG5XC8H4
    ```

>     ~> **Note:** Due to AWS eventual consistency, after calling this endpoint,
>     subsequent calls from Vault to AWS may fail for a few seconds until AWS
>     becomes consistent again.  See the [AWS secrets engine API](/api/secret/aws#rotate-root-iam-credentials) for further information on rotate-root functionality.

Reviewer: ```suggestion
<Note>
  Calls from Vault to AWS may fail immediately after calling
  `aws/config/rotate-root` until AWS becomes consistent again.
  Refer to the
  <a href="/api/secret/aws#rotate-root-iam-credentials">AWS secrets engine API</a>
  reference for additional information on rotating IAM credentials.
</Note>
```

style correction: use new aside notation (`<Note>`) and active voice.

security_token 	AQoDYXdzEEwasAKwQyZUtZaCjVNDiXXXXXXXXgUgBBVUUbSyujLjsw6jYzboOQ89vUVIehUw/9MreAifXFmfdbjTr3g6zc0me9M+dB95DyhetFItX5QThw0lEsVQWSiIeIotGmg7mjT1//e7CJc4LpxbW707loFX1TYD1ilNnblEsIBKGlRNXZ+QJdguY4VkzXxv2urxIH0Sl14xtqsRPboV7eYruSEZlAuP3FLmqFbmA0AFPCT37cLf/vUHinSbvw49C4c9WQLH7CeFPhDub7/rub/QU/lCjjJ43IqIRo9jYgcEvvdRkQSt70zO8moGCc7pFvmL7XGhISegQpEzudErTE/PdhjlGpAKGR3d5qKrHpPYK/k480wk1Ai/t1dTa/8/3jUYTUeIkaJpNBnupQt7qoaXXXXXXXXXX
```

### STS Session Tokens

> ~> **Notice:** Due to limitations in AWS, in order to use the `session_token`
> credential type, Vault **must** be configured with IAM user credentials. AWS
> does not allow temporary credentials (such as those from an IAM instance
> profile) to be used.

Reviewer: ```suggestion
<Important>
  AWS does not allow temporary credentials like those from an IAM instance
  profile. To use session tokens with Vault and AWS, you must configure Vault
  to use IAM user credentials.
</Important>
```

style correction: use new aside notation, mark the aside as important, and use active voice

> An STS session token inherits the exact same set of permissions which are
> granted to the `aws/config/root` credentials.

Reviewer: ```suggestion
STS session tokens inherit whatever permissions are granted to the `aws/config/root` credentials.
```
style correction: active voice

> A `root_access` role would then assign an inline policy with the same `ec2:*`
> permissions.

Reviewer: ```suggestion
Then the `root_access` role assigns an inline policy with the same `ec2:*` permissions.
```
style correction: active voice

```shell-session
$ vault write aws/roles/root_access \
    credential_type=session_token
```

> To generate a new set of STS federation token credentials, we simply write to
> the role using the aws/creds endpoint:

Reviewer: ```suggestion
To generate a new set of STS federation token credentials, write to the `root_access` role using the `aws/creds` endpoint:
```

style correction: accessibility, compassionate lanugage
